sdks/java/core/src/test/java/org/apache/beam/sdk/runners/PipelineRunnerTest.java

```diff
@@ -39,4 +57,50 @@ public void testInstantiation() {
     PipelineRunner<?> runner = PipelineRunner.fromOptions(options);
     assertTrue(runner instanceof CrashingRunner);
   }
+
+  private static class ScaleFn<T extends Number>
+      extends SimpleFunction<T, Double> {
+
+    private final double scalar;
+    private final Counter counter;
+
+    public ScaleFn(double scalar, Counter counter) {
+      this.scalar = scalar;
+      this.counter = counter;
+    }
+
+    @Override
+    public Double apply(T input) {
+      counter.inc();
+      return scalar * input.doubleValue();
+    }
+  }
+
+  @Test
+  @Category(ValidatesRunner.class)
+  public void testRunPTransform() {
+    final String namespace = PipelineRunnerTest.class.getName();
+    final Counter counter = Metrics.counter(namespace, "count");
+    final PipelineResult result = PipelineRunner.create().run(
+        new PTransform<PBegin, POutput>() {
+          @Override
+          public POutput expand(PBegin input) {
+            PCollection<Double> output = input
+                .apply(Create.<Integer>of(1, 2, 3, 4))
+                .apply("ScaleByTwo", MapElements.via(new ScaleFn<Integer>(2.0, counter)));
+            PAssert.that(output).containsInAnyOrder(2.0, 4.0, 6.0, 8.0);
+            return output;
+          }
+        }
+    );
+
+    assertThat(
```

Comment:
Add a comment that the correct counters are a pretty good indication that the pipeline actually ran.